### PR TITLE
Add spatial positional embedding to the beginning of FNO by default

### PR DESCRIPTION
**Pull request by dhpitt**
Also updates scripts/`DefaultDataProcessor`'s default options

### CODE REVIEW

File: neuralop/models/fno.py [hunk 1]
```diff
@@ -152,6 +164,17 @@ def __init__(
         self.preactivation = preactivation
         self.fno_block_precision = fno_block_precision
 
+        if spatial_pos_embed == "grid":
```
Comment (JeanKossaifi) [inline neuralop/models/fno.py:167]: spatial_pos_embed seems redundant, why not just `positional_embedding`?

File: config/navier_stokes_config.yaml [hunk 2]
```diff
@@ -66,7 +66,7 @@ default: &DEFAULT
     n_tests: [2000, 1000] #, 1000]
     test_resolutions: [128, 1024] #, 1024] 
     test_batch_sizes: [16, 4] #, 1]
-    positional_encoding: True
+    positional_encoding: False
```
Comment (JeanKossaifi) [inline config/navier_stokes_config.yaml:69]: Why was this changed to False?

**Review by JeanKossaifi — COMMENTED** [review]
We should be consistent between encoding and embedding, and have only one param: if we move it to the model then let's remove the dataset one to avoid confusion.

For the API I would just have one argument that is either a string (e.g. grid which is then used with default params) or an actual module.

File: neuralop/models/fno.py (hunk 1 above)
Comment (dhpitt) [reply neuralop/models/fno.py:167]: In case we add embedding in Fourier space, to avoid ambiguity

File: config/navier_stokes_config.yaml (hunk 2 above)
Comment (dhpitt) [reply config/navier_stokes_config.yaml:69]: Because it happens in the model by default

File: neuralop/models/fno.py (hunk 1 above)
Comment (JeanKossaifi) [reply neuralop/models/fno.py:167]: Let's call it positional_embedding (or encoding) it will be clearer. Fourier space embedding is still being tested and can be named differently.

File: config/navier_stokes_config.yaml (hunk 2 above)
Comment (JeanKossaifi) [reply config/navier_stokes_config.yaml:69]: As I mentioned in the main comments, then let's remove that option alltogether -- there should be only one way to do it

**Review by JeanKossaifi — COMMENTED** [review]
Responded  in the code

File: neuralop/models/fno.py (hunk 1 above)
Comment (dhpitt) [reply neuralop/models/fno.py:167]: Sounds good

**Comment by JeanKossaifi** [comment]
We should make the regular grid the default to match previous version

**Comment by JeanKossaifi** [comment]
It is much cleaner with this PR, thanks @dhpitt!
